run/image-processing/src/main/java/com/example/cloudrun/ImageMagick.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package com.example.cloudrun;
+
+// [START run_imageproc_handler_setup]
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Feature.Type;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.ImageSource;
+import com.google.cloud.vision.v1.SafeSearchAnnotation;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ImageMagick {
+
+  private static final String BLURRED_BUCKET_NAME = System.getenv("BLURRED_BUCKET_NAME");
+  private static Storage storage = StorageOptions.getDefaultInstance().getService();
+  // [END run_imageproc_handler_setup]
+
+  // [START run_imageproc_handler_analyze]
+  // Blurs uploaded images that are flagged as Adult or Violence.
+  public static void blurOffensiveImages(JsonObject data) {
+    String fileName = data.get("name").getAsString();
+    String bucketName = data.get("bucket").getAsString();
```

Comment:
Since this is sent from users and you are going to use it, should you be doing any kind of sanitizing?

Reply:
This is from a pub/sub message. The PubSubController should intercept invalid messages.